docs/src/test.md

```diff
@@ -46,6 +46,96 @@ pytest -svv forge/test/mlir/test_ops.py::test_add
 
 > - The `-svv` flag is optional and used to display more information about the test run.
 
+## Using `@pytest.mark.parametrize`
+
+The `@pytest.mark.parametrize` decorator allows you to run a single test function with multiple sets of inputs. This is particularly useful for testing a function with various combinations of arguments and expected results.
+
+### Syntax
+
+```python
+@pytest.mark.parametrize("arg1, arg2, expected", [
+    (1, 2, 3),
+    (2, 3, 5),
+    (3, 5, 8),
+])
+def test_addition(arg1, arg2, expected):
+    assert arg1 + arg2 == expected
+```
+
+### Explanation
+- The first argument is a comma-separated string of parameter names.
+- The second argument is a list of tuples, each containing values for the parameters.
+- Pytest runs the test function once for each tuple, substituting the parameter values.
+
+## Marking Specific Parameters with `pytest.param`
+
+You can use `pytest.param` to mark specific parameter combinations with additional metadata, such as expected failures (`xfail`).
+
+#### Syntax
+
+```python
+@pytest.mark.parametrize("inputs", [
+    pytest.param(
+        ((1, 2, 3), (4, 5, 6)), marks=pytest.mark.xfail(reason="reason"))
+])
+```
+
+#### Explanation
+- `pytest.param` allows you to add specific markers or metadata to individual parameter sets.
+- In this example, the first parameter combination is marked as `xfail` with a reason provided, indicating it is expected to fail.
+- This is useful when only some parameter sets are failing or not working correctly.
+
+## Skipping Tests with `@pytest.mark.skip`
+
+Use the `@pytest.mark.skip` decorator to skip a test that is not relevant or is causing issues such as segmentation faults or breaking the CI/CD pipeline.
+
+### Syntax
+
+```python
+@pytest.mark.skip(reason="Causes segmentation fault")
+def test_future_feature():
+    assert some_function() == "expected result"
+```
+
+### Explanation
+- The `reason` parameter explains why the test is being skipped.
+- Skipped tests will appear in the pytest output with the reason.
+- Skipping tests is particularly useful when a test is causing crashes (e.g., segmentation faults) or breaking the CI pipeline.
+
+
+## Marking Tests as Expected to Fail with `@pytest.mark.xfail`
```

Comment:
For xfail, it'll be good to mention what not to do as well. E.g. not using as part of test, and why :))